src/winstanley/WdlAnnotator.scala

```diff
@@ -69,10 +71,81 @@ class WdlAnnotator extends Annotator {
         annotationHolder.createErrorAnnotation(wildcardOutput, "Declaration style outputs are required in WDL 1.0 (draft 3) and later")
       }
 
+    case callBlock: WdlCallBlock =>
+      def taskForCall(callBlock: WdlCallBlock): Option[WdlTaskBlockImpl] = {
+        val taskName = callBlock.getCallableLookup.getName
+        val tasksInScope: Set[WdlTaskBlockImpl] = callBlock.findTasksInScope
+
+        tasksInScope.find(_.getTaskDeclaration.getName == taskName)
+      }
+
+      taskForCall(callBlock) match {
+        case Some(task) =>
+          // The "type" of a WdlTaskSection is determined by which one of its getXXX() methods returns not-null. Checkmate, Scala nerds.
+          val maybeTaskInputSection: Option[WdlTaskSection] = task.getTaskSectionList.asScala.find(_.getInputBlock != null)
+
+          val maybeCallInput: Option[WdlCallInput] = Option(callBlock.getCallInput)
+
+          (maybeCallInput, maybeTaskInputSection) match {
+            case (Some(callInput), Some(taskInputSection)) =>
+              compareInputListsForCall(psiElement, annotationHolder, callInput, taskInputSection)
+            case (Some(_), None) =>
+              annotationHolder.createErrorAnnotation(psiElement, "Task does not take inputs.")
+            case (None, Some(taskInputSection)) =>
+              val missingArgs = taskInputSection.getInputBlock.getDeclarationList.asScala.map(_.getName)
+              annotationHolder.createWeakWarningAnnotation(
+                psiElement,
+                "Unsupplied input(s) " + missingArgs.map("\'" + _ + "\'").mkString(", ") + " must be assigned here or, if this is the root workflow, provided in the inputs JSON."
+              )
+            case (None, None) => ()
+          }
+
+        case None => ()
+      }
     case _ => ()
   }
 
   private def kvName(kv: WdlKv): Option[String] = Option(kv.getNode.findChildByType(WdlTypes.IDENTIFIER)).map(_.getText)
   private def runtimeKeyword(r: WdlRuntimeBlock): Option[ASTNode] = Option(r.getNode.findChildByType(WdlTypes.RUNTIME))
   private def outputKeyword(r: WdlTaskOutputs): Option[ASTNode] = Option(r.getNode.findChildByType(WdlTypes.OUTPUT))
+
+  private def compareInputListsForCall(psiElement: PsiElement, annotationHolder: AnnotationHolder, callInput: WdlCallInput, taskInputSection: WdlTaskSection) = {
+    // In the interest of unambiguous naming, this method (ab)uses the formal definitions of parameter (declared on method) and argument (passed to method)
```

Comment:
I like this distinction... to make this just as clear in the variable names could they be, for example `taskParameters` and `callArguments`?